tools/ci_build/github/azure-pipelines/templates/win-ci-vs-2019.yml

```diff
@@ -119,13 +126,47 @@ jobs:
     workingDirectory: '$(Build.BinariesDirectory)'
     displayName: 'Install python modules'
 
+  - powershell: |
+      if ([string]::IsNullOrEmpty((Get-Command ccache -errorAction SilentlyContinue)))
+      {
+        choco install ccache -y --version 4.7.4
+        $ccache_path = (Get-Command ccache).Source
+        $ccache_parent_dir = (Split-Path -parent $ccache_path)
+        Copy-Item "C:\ProgramData\chocolatey\lib\ccache\tools\ccache-4.7.4-windows-x86_64\ccache.exe" -Destination "C:\ProgramData\chocolatey\bin\cl.exe"
+        Get-ChildItem $ccache_parent_dir
+        ccache --version
+      }
+    displayName: Install ccache and update PATH to use linked versions of gcc, cc, etc
+    condition: eq(${{ parameters.WITH_CACHE }}, true)
+
+  - task: Cache@2
+    inputs:
+      key: '"$(TODAY)" | ccache | "$(System.StageName)" | "$(Build.SourceBranch)" | "$(Build.SourceVersion)" '
+      path: $(CCACHE_DIR)
+      restoreKeys: |
+        "$(TODAY)" | ccache | "$(System.StageName)" | "$(Build.SourceBranch)"
+        "$(TODAY)" | ccache | "$(System.StageName)"
+        "$(TODAY)" | ccache |
+    displayName: Cache Task
+    condition: eq(${{ parameters.WITH_CACHE }}, true)
+
   - ${{ if or(eq(parameters.RunOnnxRuntimeTests, true), eq(parameters.GenerateDocumentation, true)) }}:
       - task: PowerShell@2
         displayName: 'Install ONNX'
         inputs:
           filePath: '$(Build.SourcesDirectory)/tools/ci_build/github/windows/install_third_party_deps.ps1'
           workingDirectory: '$(Build.BinariesDirectory)'
-          arguments: -cpu_arch ${{ parameters.buildArch }} -install_prefix $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\installed -build_config ${{ parameters.BuildConfig }}
+          ${{ if eq(parameters.WITH_CACHE, true) }}:
+            arguments: -cpu_arch ${{ parameters.buildArch }} -install_prefix $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\installed -build_config ${{ parameters.BuildConfig }} -use_cache
+          ${{ else }}:
+            arguments: -cpu_arch ${{ parameters.buildArch }} -install_prefix $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\installed -build_config ${{ parameters.BuildConfig }}
```

Comment:
instead of cut-and-pasting a long list of arguments, can we instead just append `--use_cache` if WITH_CACHE is true? that's much easier to maintain if any of the other arguments need to change in the future.